thunx-autoconfigure/src/main/java/eu/xenit/contentcloud/thunx/gateway/autoconfigure/GatewayAutoConfiguration.java

```diff
@@ -0,0 +1,51 @@
+package eu.xenit.contentcloud.thunx.gateway.autoconfigure;
+
+import eu.xenit.contentcloud.opa.client.OpaClient;
+import eu.xenit.contentcloud.opa.client.rest.RestClientConfiguration;
+import eu.xenit.contentcloud.thunx.pdp.PolicyDecisionComponentImpl;
+import eu.xenit.contentcloud.thunx.pdp.PolicyDecisionPointClient;
+import eu.xenit.contentcloud.thunx.pdp.opa.OpenPolicyAgentPDPClient;
+import eu.xenit.contentcloud.thunx.spring.gateway.filter.AbacGatewayFilterFactory;
+import eu.xenit.contentcloud.thunx.spring.security.ReactivePolicyAuthorizationManager;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gateway.filter.factory.AbstractGatewayFilterFactory;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authorization.ReactiveAuthorizationManager;
+import org.springframework.security.web.server.authorization.AuthorizationContext;
+
+@Configuration
+@ConditionalOnClass({OpaClient.class, AbstractGatewayFilterFactory.class})
+@EnableConfigurationProperties(OpaProperties.class)
+public class GatewayAutoConfiguration {
+
+    @Bean
+    @ConditionalOnMissingBean
+    public OpaClient opaClient(OpaProperties opaProperties) {
+        return OpaClient.builder()
+                .httpLogging(RestClientConfiguration.LogSpecification::all)
+                .url(opaProperties.getUrl())
+                .build();
+    }
+
+    @Bean
+    @ConditionalOnMissingBean
+    public PolicyDecisionPointClient pdpClient(OpaClient opaClient) {
+        return new OpenPolicyAgentPDPClient(opaClient, request -> "data.vfinance.brokercontent.allow == true");
```

Comment:
The OPA query here is very specific.
It should either be configurable with some property, or else this bean can't really be autoconfigured at all.

Reply:
True, lets fix that by introducing the `OpaQueryProvider` as a dependency :+1: